src/core/lib/json/json.h

```diff
@@ -86,140 +86,107 @@ class Json {
 
   // Construct by moving a string.
   // NOLINTNEXTLINE(google-explicit-constructor)
-  Json(std::string&& string)
-      : type_(Type::kString), string_value_(std::move(string)) {}
+  Json(std::string&& string) : value_(Value(std::move(string))) {}
   Json& operator=(std::string&& string) {
-    type_ = Type::kString;
-    string_value_ = std::move(string);
+    value_ = Value(std::move(string));
     return *this;
   }
 
   // Construct from bool.
   // NOLINTNEXTLINE(google-explicit-constructor)
-  Json(bool b) : type_(b ? Type::kTrue : Type::kFalse) {}
+  Json(bool b) : value_(b) {}
   Json& operator=(bool b) {
-    type_ = b ? Type::kTrue : Type::kFalse;
+    value_ = b;
     return *this;
   }
 
   // Construct from any numeric type.
   template <typename NumericType>
   // NOLINTNEXTLINE(google-explicit-constructor)
-  Json(NumericType number)
```

Comment:
For a follow-up PR (I didn't catch this earlier, sorry!)

This function provides implicit conversion for anything that `std::to_string` supports - which may change in the future and bite us... would suggest making an explicit set of types for a public API.

Worse though:
```
std::to_string relies on the current locale for formatting purposes, and therefore concurrent calls to std::to_string from multiple threads may result in partial serialization of calls.
```
(source: https://en.cppreference.com/w/cpp/string/basic_string/to_string)

I'm not terribly fussed about the serialization (scary, but not terrible) - the locale dependency is worrisome though - I don't want to be dealing with EU specific bugs in gRPC ever.

Reply:
Good catch!  I will fix this in a subsequent PR.